modules/segger/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+if(CONFIG_USE_SEGGER_RTT)
+  set(SEGGER_DIR ${ZEPHYR_CURRENT_MODULE_DIR})
+  zephyr_include_directories_ifdef(CONFIG_USE_SEGGER_RTT
+    ${SEGGER_DIR}/SEGGER
+    ${SEGGER_DIR}/Config
+    .
+    )
+  zephyr_sources_ifdef(CONFIG_USE_SEGGER_RTT
```

Comment:
`CONFIG_USE_SEGGER_RTT` is already being checked in line 1.
So no need for the `ifdef` here.

Now the the files are being moved, should we use `zephyr_library()` and `zephyr_library_sources` here, so we don't blindly put those files into `libzephyr.a`.